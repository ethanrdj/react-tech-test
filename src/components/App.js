import React from "react";
import "../styles/App.css";
import Search from "./Search";
import { useState } from "react";

function App() {
  const [searchResults, setSearchResults] = useState("");
  return (
    <div className="App" data-testid="App-test">
      <img
        className="nasa-logo"
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        alt="logo"
        data-testid="image-test"
      ></img>
      <Search setSearchResults={setSearchResults} />
    </div>
  );
}

export default App;
