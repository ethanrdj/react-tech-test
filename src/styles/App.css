.App {
  display: flex;
  flex-direction: column;
}

.nasa-logo {
  height: 40vh;
  width: 40vw;
}
