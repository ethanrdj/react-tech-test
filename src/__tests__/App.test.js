import React from "react";
import { render } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  const { asFragment } = render(<App />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("ensures the component mounts", () => {
    const { getAllByTestId } = render(<App />);

    expect(getAllByTestId("App-test")).toBeTruthy();
    expect(getAllByTestId("image-test")).toBeTruthy();
  });
});
