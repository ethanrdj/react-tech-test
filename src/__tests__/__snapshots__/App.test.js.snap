// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
    data-testid="App-test"
  >
    <img
      alt="logo"
      class="nasa-logo"
      data-testid="image-test"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <input
      class="search-bar"
      data-testid="search-test"
      type="text"
    />
  </div>
</DocumentFragment>
`;
